--- ../xerces-c-3.2.2/src/xercesc/util/Transcoders/IconvGNU/IconvGNUTransService.hpp	2018-02-14 19:22:38.000000000 -0600
+++ xerces_source/src/xercesc/util/Transcoders/IconvGNU/IconvGNUTransService.hpp	2019-02-06 00:09:04.707945899 -0600
@@ -25,7 +25,7 @@
 #include <xercesc/util/TransService.hpp>
 #include <xercesc/util/Mutexes.hpp>
 
-#include <iconv.h>
+#include ICONV_INCLUDE
 
 XERCES_CPP_NAMESPACE_BEGIN
 
