--- ../xerces-c-3.2.2/src/CMakeLists.txt	2018-05-01 18:58:20.000000000 -0500
+++ xerces_source/src/CMakeLists.txt	2019-02-05 23:38:57.832510132 -0600
@@ -1271,6 +1271,10 @@
 add_library(xerces-c
   ${libxerces_c_SOURCES}
   ${libxerces_c_RESOURCES})
+
+target_include_directories(xerces-c BEFORE PUBLIC "$ENV{GNU_ICONV_ROOT}/include")
+list(APPEND libxerces_c_DEPS "$ENV{GNU_ICONV_ROOT}/lib/libiconv.so")
+
 target_link_libraries(xerces-c ${libxerces_c_DEPS})
 if(XERCES_USE_NETACCESSOR_CURL)
   target_include_directories(xerces-c SYSTEM PRIVATE ${CURL_INCLUDE_DIRS})
